crates/miden-testing/src/kernel_tests/tx/test_account_interface.rs

```diff
@@ -75,9 +75,11 @@ async fn check_note_consumability_well_known_notes_success() -> anyhow::Result<(
 
     assert_matches!(execution_check_result, NoteConsumptionInfo { successful, failed, .. } => {
         assert_eq!(successful.len(), notes.len());
-        successful.iter().zip(notes.iter()).for_each(|(success, note)| {
-            assert_eq!(success, note);
-        });
+
+        // we asserted that `successful` and `notes` vectors have the same length, so it's safe to
+        // check their equality that way
+        successful.iter().for_each(|successful_note| assert!(notes.contains(successful_note)));
```

Comment:
This rewritten assertion is less strict than the one before. The `zip` version asserts that success and input notes are in the same order, while the rewritten one only asserts that success notes and input notes are the same set, but not necessarily the same order.

I don't recall whether this stricter assertion is necessary, but what was the reason for changing it here?

Reply:
The reason is that the order of notes in `successful` vector is not fixed (here is more detailed explanation: https://github.com/0xMiden/miden-base/pull/1776#discussion_r2293424297), so I have to update the ordering of the `notes` vector each time the hashes of the kernel procedures are changed. I'm not 100% sure whether the ordering is crucial here, but I now for sure that we can't (at least for now) guarantee the ordering of the notes which are generated during the build of the `TransactionContext`.